nautobot/core/api/views.py

```diff
@@ -773,38 +776,44 @@ def execute_graphql_request(self, request, data, query, variables, operation_nam
         if not query:
             raise HttpError(HttpResponseBadRequest("Must provide query string."))
 
+        schema = self.schema.graphql_schema
+
+        schema_validation_errors = validate_schema(schema)
+        if schema_validation_errors:
+            return ExecutionResult(data=None, errors=schema_validation_errors)
+
         try:
-            backend = self.get_backend(request)
-            document = backend.document_from_string(self.graphql_schema, query)
+            document = parse(query)
         except Exception as e:
-            return ExecutionResult(errors=[e], invalid=True)
+            return ExecutionResult(errors=[e])
+
+        operation_ast = get_operation_ast(document, operation_name)
 
-        operation_type = document.get_operation_type(operation_name)
-        if operation_type and operation_type != "query":
+        if operation_ast is not None and operation_ast.operation != OperationType.QUERY:
             raise HttpError(
-                HttpResponseBadRequest(f"'{operation_type}' is not a supported operation, Only query are supported.")
+                HttpResponseBadRequest(
+                    f"'{operation_ast.operation}' is not a supported operation, Only query are supported."
+                )
             )
 
-        try:
-            extra_options = {}
-            if self.executor:
-                # We only include it optionally since
-                # executor is not a valid argument in all backends
-                extra_options["executor"] = self.executor
+        validation_errors = validate(schema, document, self.validation_rules, graphene_settings.MAX_VALIDATION_ERRORS)
+        if validation_errors:
+            return ExecutionResult(data=None, errors=validation_errors)
```

Comment:
Possibly could omit this addition (copied in from the updated graphene-django base view) as we aren't currently setting any `validation_rules`.